### PR TITLE
Multiple Enhancements

### PR DESCRIPTION
**Pull request by donavanbecker**
- Fix for battery characteristic not an optional characteristic
- Update to use Homebridge 1.3.0 onSet and onGet
- Add passowrd schema form
- Update Characteristics

### CODE REVIEW

**Review by donavanbecker — COMMENTED** [review]
reverted a few changes as asked.

File: src/myq-garagedoor.ts [hunk 1]
```diff
@@ -168,49 +182,16 @@ export class myQGarageDoor extends myQAccessory {
...
-
   // Open or close the garage door.
-  private setDoorState(value: CharacteristicValue, callback?: CharacteristicSetCallback): boolean {
+  private setDoorState(value: CharacteristicValue): boolean {
 
     const myQState = this.doorStatus();
     const accessory = this.accessory;
     const hap = this.hap;
 
     if(myQState === -1) {
```
Comment (donavanbecker) [inline src/myq-garagedoor.ts:192]: not sure how you want to handle this?

File: src/myq-garagedoor.ts [hunk 2]
```diff
@@ -224,12 +205,8 @@ export class myQGarageDoor extends myQAccessory {
       this.log.error("%s: Unable to %s door while currently attempting to complete %s. myQ must complete it's existing action before attempting a new one.",
         accessory.displayName, actionAttempt, actionExisting);
 
-      if(callback) {
-        callback(new Error("Unable to accept a new set event while another is completing."));
-      }
-
+      //new Error("Unable to accept a new set event while another is completing."));
```
Comment (donavanbecker) [inline src/myq-garagedoor.ts:208]: not sure how you want to handle this?

File: src/myq-garagedoor.ts (hunk 1 above)
Comment (donavanbecker) [inline src/myq-garagedoor.ts:192]: not sure how you want to handle this?

File: src/myq-garagedoor.ts [hunk 3]
```diff
@@ -272,23 +244,14 @@ export class myQGarageDoor extends myQAccessory {
...
     }
 
     // HomeKit has told us something that we don't know how to handle.
     this.log.error("%s: Unknown SET event received: %s.", accessory.displayName, value);
 
-    if(callback) {
-      callback(new Error("Unknown SET event received: " + value.toString()));
-    }
-
+    //new Error("Unknown SET event received: " + value.toString()));
```
Comment (donavanbecker) [inline src/myq-garagedoor.ts:253]: not sure how you want to handle this?

Comment (hjdhjd) [reply src/myq-garagedoor.ts:253]: Stick to logging the error and let it keep running. This code path should never execute anyway...I've never seen it happen in all these years. :smile:

File: src/myq-garagedoor.ts (hunk 2 above)
Comment (hjdhjd) [reply src/myq-garagedoor.ts:208]: Same as below...log the error, call it a day.

Comment (hjdhjd) [reply src/myq-garagedoor.ts:208]: Ditto...it already returns false if it errors out.

**Pull request merged**